nncf/experimental/torch2/function_hook/nncf_graph/nncf_graph_builder.py

```diff
@@ -77,20 +79,86 @@ def get_dtype(dtype: torch.dtype) -> Dtype:
     :param meta: The metadata associated with the node.
     :return: The PTOperatorMetatype object.
     """
-    node_metatype = cast(om.PTOperatorMetatype, om.PT_OPERATOR_METATYPES.get_operator_metatype_by_op_name(node_type))
-    node_sub_meta_type: Optional[om.PTOperatorMetatype] = None
+    node_metatype = cast(
+        type[om.PTOperatorMetatype], om.PT_OPERATOR_METATYPES.get_operator_metatype_by_op_name(node_type)
+    )
+    node_sub_meta_type: Optional[type[om.PTOperatorMetatype]] = None
     if node_metatype.get_subtypes() and isinstance(meta, FunctionMeta):
         node_sub_meta_type = node_metatype.determine_subtype(function_args=meta.args, functions_kwargs=meta.kwargs)
     return node_sub_meta_type or node_metatype
 
 
+def is_constant_input_node(nx_graph: nx.MultiDiGraph, node: int) -> bool:
+    """
+    Check if a node is a constant input node or constant subgraph:
+
+    1) constant
+    2) quantize_function -> constant
+
+    :param nx_graph: The graph to check the node from.
+    :param node: The node to check.
+    :return: True if the node is a constant input node, False otherwise.
+    """
+    meta = nx_graph.nodes[node]["meta"]
+
+    # 1) Input node is a constant node (parameter or buffer)
+    if isinstance(meta, ConstMeta):
+        return True
+
+    # 2) Quantize node with constant input
+    if (
+        isinstance(meta, FunctionMeta)
+        and meta.func_name in om.QUANTIZE_NODE_TYPES
+        and isinstance(nx_graph.nodes[node]["meta"], FunctionMeta)
+    ):
+        return all(isinstance(nx_graph.nodes[s_node]["meta"], ConstMeta) for s_node, _ in nx_graph.in_edges(node))
+
+    return False
+
+
+def get_constant_port_ids(nx_graph: nx.MultiDiGraph, node: int) -> Set[int]:
+    """
+    Get the indices of input ports corresponding to the constant node or subgraph.
+
+    :param nx_graph: The graph to get the constant port IDs from.
+    :param node: The node to get the constant port IDs from.
+    :return: The list of input port indices with constants.
+    """
+    constant_port_ids: Set[int] = set()
+
+    for s_node, _, data in nx_graph.in_edges(node, data=True):
+        if is_constant_input_node(nx_graph, s_node):
+            meta = cast(EdgeMeta, data["meta"])
+            constant_port_ids.add(meta.input_port)
+
+    return constant_port_ids
+
+
+def get_layer_attributes(
+    nx_graph: nx.MultiDiGraph, node: int, meta: Union[ConstMeta, FunctionMeta, InOutMeta]
+) -> Optional[BaseLayerAttributes]:
+    """
+    Get the layer attributes of a node in the graph.
+
+    :param nx_graph: The graph to get the layer attributes from.
+    :param node: The node to get the layer attributes from.
+    :param meta: The metadata associated with the node.
+    :return: The layer attributes of the node.
+    """
+    if isinstance(meta, FunctionMeta):
+        constant_port_ids = get_constant_port_ids(nx_graph, node)
+        return PT2OpLayerAttributes(meta.func, meta.args, meta.kwargs, constant_port_ids)
```

Comment:
Why do we need both `constant_port_ids` in `PT2OpLayerAttributes and `weights_port_id` in the metatypes? https://github.com/openvinotoolkit/nncf/blob/develop/nncf/torch/graph/operator_metatypes.py#L207

Reply:
`constant_port_ids` collects all constanat ports like weights and bias.
`weights_port_id` needs to find only weight.

Reply:
Ok, there are models with constant subgraphs which contain linear layers (swin_v2), does you approach cover such cases?

Reply:
All 3 inputs marks as constants